### PR TITLE
Fix issue with FastAPI mounts.

### PR DESCRIPTION
**Pull request by nsthorat**

### CODE REVIEW

File: lilac/server.py [hunk 1]
```diff
@@ -98,7 +98,11 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
     url = URL(scope=scope).path
     root_path = scope.get('root_path') or ''
     ends_with_slash = (
-      url.endswith('/') and url != '/' and root_path and not url.startswith(root_path + '/api')
```
Comment (brilee) [inline lilac/server.py:101]: can we make this a unit test to enumerate the kinds of URLs that this does/doesn't accept?

**Review by brilee — APPROVED** [review]

**Review by dsmilkov — APPROVED** [review]

File: integration_tests/fastapi_mount_test.sh [hunk 2]
```diff
@@ -24,6 +24,5 @@ until curl --fail --silent "$URL" > /dev/null; do
 done
 
 
-curl --fail --silent "http://localhost:$PORT/lilac_sub/api/v1/tasks/"
-
-curl --fail --silent "http://localhost:$PORT/lilac_sub/"
+curl --fail --silent "http://localhost:$PORT/$MOUNT_PATH/api/v1/tasks/" > /dev/null
+curl --fail --silent "http://localhost:$PORT/$MOUNT_PATH/" --max-redirs 0 > /dev/null
```
Comment (derek-pyne) [inline integration_tests/fastapi_mount_test.sh:28]:
```suggestion
curl --fail --silent "http://localhost:$PORT/$MOUNT_PATH/" -L --max-redirs 0 > /dev/null
```

Need to add `-L` (follow redirects) for `--max-redirs` to have any effect

File: lilac/server.py (hunk 1 above)
Comment (nsthorat) [reply lilac/server.py:101]: done

File: integration_tests/fastapi_mount_test.sh (hunk 2 above)
Comment (nsthorat) [reply integration_tests/fastapi_mount_test.sh]: Thanks! Done :)

Comment (nsthorat) [reply integration_tests/fastapi_mount_test.sh]: I had to migrate this to a python test, FYI.

**Pull request merged**